pkg/registry/client.go

```diff
@@ -195,12 +190,9 @@ func ClientOptPlainHTTP() ClientOption {
 	}
 }
 
-// ClientOptResolver returns a function that sets the resolver setting on a client options set
-func ClientOptResolver(resolver remotes.Resolver) ClientOption {
-	return func(client *Client) {
-		client.resolver = func(ref registry.Reference) (remotes.Resolver, error) {
-			return resolver, nil
-		}
+func ClientOptResolver(_ remotes.Resolver) ClientOption {
+	return func(c *Client) {
+		c.err = errDeprecatedRemote
```

Comment:
This is still a breaking change and requires a Helm major version change. It's about taking functions and making them non-functional with an error.

Reply:
As previously noted support for containerd was removed from oras-go and there are no plans to reintroduce it. See https://github.com/oras-project/oras-go/issues/64#issuecomment-1077845399 for details.

I have yet to see any suggestions on how to get this addressed, no plans on the next major version of Helm either. Please do share any insights that you are privy to.

Frankly I do not see a way forward with this and will limit my effort going forward. Anyone is welcome to pick this up and continue the work.

Reply:
For the records, this client option:
1. was recently added in [3.13.0](https://github.com/helm/helm/releases/tag/v3.13.0) (under 6 months ago, and only two weeks before this PR was opened)
2. was added [without any real testing](https://github.com/helm/helm/commit/80bc7df7820513e79fc24a1d4bbffd96ef6a9adf)
3. was in beta / RC1 for less than 10 days ([3.13.0-rc.1](https://github.com/helm/helm/releases/tag/v3.13.0-rc.1) is [identical](https://github.com/helm/helm/compare/v3.13.0-rc.1...v3.13.0) to 3.13.0)
4. was [implemented](https://github.com/helm/helm/pull/11129) by the same person that [opened the issue](https://github.com/helm/helm/issues/10623)
5. has what seems like 1 known use across [all of github's public codebases](https://github.com/search?q=ClientOptResolver+NOT+path%3A%2F.*test.*%2F+language%3Ago+NOT+%22func+ClientOptResolver%22&type=code), barring matches that seem to be forks or vendored modules
    - [vmware-labs/distribution-tooling-for-helm](https://github.com/vmware-labs/distribution-tooling-for-helm/blob/9ef5afa4a24d76e08342a9324acb3fb7c9deb181/pkg/artifacts/helm.go#L119) does use the current implementation, but again, VMware was the team that opened the issue and implemented this option in helm.
    - [d2iq-labs/csi-driver-trusted-ca](https://github.com/d2iq-labs/csi-driver-trusted-ca/blob/ce15c6710d71963270bda57b71166eb6da359ae5/pkg/source/oci_source.go#L55)'s usage is of it's own identically named option. This repository [depends on helm 3.11.1](https://github.com/d2iq-labs/csi-driver-trusted-ca/blob/ce15c6710d71963270bda57b71166eb6da359ae5/go.mod#L28) and has seemingly implemented [#10623](https://github.com/helm/helm/issues/10623) itself
7. has alternative solutions to what the original issue was asking for -- albeit yes, breaking

In terms of breaking changes, I think this is very manageable and the benefits of moving to oras v2 outweigh the costs of delaying this PR until the theoretical helm v4 release. I'm fairly sure that VMware team members @antgamdia or @alemorcuq would be okay with managing this change, as they are the source of this API in helm and still seem to be the only people affected.

Reply:
Great analysis @zevisert! Just FYI https://github.com/d2iq-labs/csi-driver-trusted-ca is not maintained so does not need to be considered in any discussion.

I'm also very supportive of getting this change in.

Reply:
Hey @mattfarina, I understand if you're busy with other things, but we'd appreciate if you could come back here reconsider this sometime soon :)

Reply:
I'd like to add one more thing to the discussion: This is the only remaining reference to containerd v1.x in Helm. Removing this API allows the entire containerd dependency to be dropped. This would allow Helm and the upcoming containerd 1.8 API to be used in the same Go module, avoiding ambiguous import errors (e.g. `github.com/containerd/containerd v1.7.16` vs. `github.com/containerd/containerd/api v1.8.0-rc.0`).